### PR TITLE
Object security

### LINKED ISSUE #462: [SecurityBundle] Add possibility to secure specific objects
<a href="https://github.com/danrot"></a>
By @danrot *from* https://github.com/sulu-cmf/SuluSecurityBundle/issues/82

At the moment it's only possible to secure base on the defined security contexts. However, it should be possible to secure specific objects (like a specific page or database object) as well.

### PR DESCRIPTION
**Pull request by danrot**
Adds the possibility to secure specific objects, currently also implemented for pages.

Has to be merged with https://github.com/sulu-io/sulu-standard/pull/433

__tasks:__

- [ ] test coverage
- [ ] gather feedback for my changes
- [ ] submit changes to the documentation

__informations:__

| q                | a
| ---------------- | ---
| Fixed tickets    | fixes #462 
| BC breaks        | none
| Documentation PR | TODO

### CODE REVIEW

File: src/Sulu/Bundle/ContentBundle/Admin/SuluContentContentNavigation.php [hunk 1]
```diff
@@ -21,6 +21,12 @@ public function __construct()
 
         $this->setName('Content');
 
+        $this->addNavigationItem($this->getContent());
+
+        $this->addNavigationItem($this->getSeo());
+        $this->addNavigationItem($this->getExcerpt());
+
+        $this->addNavigationItem($this->getSettings());
```
Comment (dantleech) [inline src/Sulu/Bundle/ContentBundle/Admin/SuluContentContentNavigation.php]: Why the inconsistent spacing?

File: src/Sulu/Bundle/ContentBundle/Controller/NavigationController.php [hunk 2]
```diff
@@ -25,16 +25,13 @@ class NavigationController extends Controller
...
      * @return \Symfony\Component\HttpFoundation\Response
      */
-    public function contentAction(Request $request)
+    public function contentAction()
     {
         if ($this->has(self::SERVICE_NAME)) {
             /** @var ContentNavigation $contentNavigation */
             $contentNavigation = $this->get(self::SERVICE_NAME);
-            $uuid = $request->get('uuid');
-            $contentNavigation->generate($uuid !== 'index');
```
Comment (dantleech) [inline src/Sulu/Bundle/ContentBundle/Controller/NavigationController.php]: So this no longer needs the REquest?

File: src/Sulu/Bundle/ContentBundle/Controller/NavigationController.php [hunk 3]
```diff
@@ -25,16 +25,13 @@ class NavigationController extends Controller
...
 
             return new JsonResponse($contentNavigation->toArray('content'));
         } else {
```
Comment (dantleech) [inline src/Sulu/Bundle/ContentBundle/Controller/NavigationController.php]: Not part of this PR, but `else` is not required here.

File: src/Sulu/Bundle/SecurityBundle/Controller/PermissionController.php [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+            $permissions = $this->accessControlManager->getPermissions($type, $identifier);
+
+            return $this->viewHandler->handle(View::create(
+                array(
+                    'id' => $identifier,
+                    'type' => $type,
+                    'permissions' => $permissions
+                )
+            ));
+        } catch (RestException $exc) {
```
Comment (dantleech) [inline src/Sulu/Bundle/SecurityBundle/Controller/PermissionController.php]: `s/exc/e`

File: src/Sulu/Bundle/ContentBundle/Admin/SuluContentContentNavigation.php (hunk 1 above)
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Admin/SuluContentContentNavigation.php]: Think I just copied it from a different place, will change it. Maybe it was done like this because seo and excerpt are the content extensions.

File: src/Sulu/Bundle/ContentBundle/Controller/NavigationController.php (hunk 2 above)
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Controller/NavigationController.php]: It will need it again ;-)

File: src/Sulu/Bundle/AdminBundle/Navigation/ContentNavigationItem.php [hunk 5]
```diff
@@ -241,7 +241,7 @@ public function toArray()
             'action' => $this->getAction(),
             'display' => $this->getDisplay(),
             'component' => $this->getComponent(),
-            'componentOptions' => $this->getComponentOptions(),
+            'componentOptions' => array_merge($this->getComponentOptions(), $options),
```
Comment (dantleech) [inline src/Sulu/Bundle/AdminBundle/Navigation/ContentNavigationItem.php]: Not related, I guess, to this PR - but would it make sense to stop using `->toArray` generally and use the JMS serializer? /cc @wachterjohannes

File: src/Sulu/Bundle/ContentBundle/Controller/NodeController.php [hunk 6]
```diff
@@ -535,7 +537,7 @@ public function postTriggerAction($uuid, Request $request)
                     $data = $repository->copyNode($uuid, $srcLocale, $webspace, $language, $userId);
                     break;
                 case 'order':
-                    $position = (int) $this->getRequestParameter($request, 'position', true);
+                    $position = (int)$this->getRequestParameter($request, 'position', true);
```
Comment (dantleech) [inline src/Sulu/Bundle/ContentBundle/Controller/NodeController.php]: I always have a space after a type hint

File: src/Sulu/Bundle/ContentBundle/Controller/NodeController.php [hunk 7]
```diff
@@ -582,4 +584,13 @@ public function getSecurityContext()
...
         }
     }
+
+    /**
+     * Returns the class name of the object to check
+     * @return string
+     */
+    public function getSecuredClass()
+    {
+        return Structure::class;
```
Comment (dantleech) [inline src/Sulu/Bundle/ContentBundle/Controller/NodeController.php]: Should this be coupled to the controller? Why can we not infer this from the object?

File: src/Sulu/Bundle/SecurityBundle/Command/CreateRoleCommand.php [hunk 8]
```diff
@@ -81,7 +81,7 @@ function ($value) use (&$securityContextsFlat) {
             $permission = new Permission();
             $permission->setRole($role);
             $permission->setContext($securityContext);
-            $permission->setPermissions(120);
+            $permission->setPermissions(121);
```
Comment (dantleech) [inline src/Sulu/Bundle/SecurityBundle/Command/CreateRoleCommand.php]: What does this mean?

File: src/Sulu/Bundle/SecurityBundle/Controller/PermissionController.php [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+            return $this->viewHandler->handle(View::create($exc->toArray(), 400));
+        }
```
Comment (dantleech) [inline src/Sulu/Bundle/SecurityBundle/Controller/PermissionController.php]: Why do we need to handle this? It should be handled automatically no?

File: src/Sulu/Bundle/SecurityBundle/Controller/PermissionController.php [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+                    $permission
+                );
+            }
+
+            return $this->viewHandler->handle(View::create(array(
+                'id' => $identifier,
+                'type' => $type,
+                'permissions' => $permissions
+            )));
+        } catch (RestException $exc) {
```
Comment (dantleech) [inline src/Sulu/Bundle/SecurityBundle/Controller/PermissionController.php]: `s/exc/e`

Comment (dantleech) [inline src/Sulu/Bundle/SecurityBundle/Controller/PermissionController.php]: But same question.. why do we need to do this?

File: src/Sulu/Bundle/SecurityBundle/EventListener/SuluSecurityListener.php [hunk 11]
```diff
@@ -70,10 +74,23 @@ public function onKernelController(FilterControllerEvent $event)
                 break;
         }
 
-        // find locale for check
+        $securityContext = null;
         $locale = $controller[0]->getLocale($event->getRequest());
+        $objectType = null;
+        $objectId = $request->get('id') ?: $request->get('uuid');
```
Comment (dantleech) [inline src/Sulu/Bundle/SecurityBundle/EventListener/SuluSecurityListener.php]: This should be a method in the interface.

File: src/Sulu/Bundle/SecurityBundle/Command/CreateRoleCommand.php (hunk 8 above)
Comment (dantleech) [inline src/Sulu/Bundle/SecurityBundle/Command/CreateRoleCommand.php]: The permissions are configurable, but this is hard-coded. Makes it possible for random behavior.

File: src/Sulu/Bundle/SecurityBundle/Entity/User.php [hunk 12]
```diff
@@ -39,6 +39,16 @@ class User extends BaseUser
     private $userSettings;
 
     /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        $this->userRoles = new \Doctrine\Common\Collections\ArrayCollection();
+        $this->userGroups = new \Doctrine\Common\Collections\ArrayCollection();
+        $this->userSettings = new \Doctrine\Common\Collections\ArrayCollection();
```
Comment (dantleech) [inline src/Sulu/Bundle/SecurityBundle/Entity/User.php]: use `use` instead of the FQN

File: src/Sulu/Bundle/SecurityBundle/Entity/User.php [hunk 13]
```diff
@@ -72,6 +82,21 @@ public function getUserRoles()
...
+     * {@inheritdoc}
+     */
+    public function getRoles()
+    {
+        $roles = parent::getRoles();
+
+        foreach ($this->getUserRoles() as $userRole) {
+            /** @var UserRole $userRole */
+            $roles[] = $userRole->getRole()->getIdentifier();
+        }
```
Comment (dantleech) [inline src/Sulu/Bundle/SecurityBundle/Entity/User.php]: Would there ever be a situation where you only want the "non-user" roles?

File: src/Sulu/Bundle/SecurityBundle/EventListener/SuluSecurityListener.php [hunk 14]
```diff
@@ -40,12 +42,14 @@ public function onKernelController(FilterControllerEvent $event)
     {
         $controller = $event->getController();
 
-        if (!$controller[0] instanceof SecuredControllerInterface) {
+        if (
+            !$controller[0] instanceof SecuredControllerInterface &&
```
Comment (dantleech) [inline src/Sulu/Bundle/SecurityBundle/EventListener/SuluSecurityListener.php]: We should probably assign `$conteoller[0]` to its own variable rather than use the index `0` all of the time

File: src/Sulu/Bundle/SecurityBundle/Resources/public/js/components/permission-tab/components/form/main.js [hunk 15]
```diff
@@ -0,0 +1,175 @@
...
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+define(['config', 'sulusecurity/collections/roles'], function(Config, Roles) {
+
+    'use strict';
+
+    var permissionUrl = '/admin/api/permissions',
```
Comment (dantleech) [inline src/Sulu/Bundle/SecurityBundle/Resources/public/js/components/permission-tab/components/form/main.js]: Not related. But we should really define all of these routes in Symfony and pass them as options.

File: src/Sulu/Bundle/SecurityBundle/Tests/Functional/Controller/PermissionControllerTest.php [hunk 16]
```diff
@@ -0,0 +1,175 @@
...
+    private $role;
+
+    /**
+     * @var MutableAclProviderInterface
+     */
+    private $aclProvider;
+
+    public function setUp()
+    {
+        $this->em = $this->db('ORM')->getOm();
```
Comment (dantleech) [inline src/Sulu/Bundle/SecurityBundle/Tests/Functional/Controller/PermissionControllerTest.php]: Missing member property for `$this->em`

File: src/Sulu/Component/Security/Authorization/AccessControl/AccessControlManagerInterface.php [hunk 17]
```diff
@@ -0,0 +1,37 @@
...
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Security\Authorization\AccessControl;
+
+use Sulu\Component\Security\Authentication\SecurityIdentityInterface;
+
+/**
+ * Defines methods for assigning security to objects
+ * @package Sulu\Component\Security\Authorization\AccessControl
```
Comment (dantleech) [inline src/Sulu/Component/Security/Authorization/AccessControl/AccessControlManagerInterface.php]: Remove package - is that autogenerated by phpstorm?

File: src/Sulu/Component/Security/Authorization/AccessControl/AccessControlVoter.php [hunk 18]
```diff
@@ -0,0 +1,79 @@
...
+            $logger,
+            $allowIfObjectIdentityUnavailable
+        );
+
+        $this->aclProvider = $aclProvider;
+    }
+
+    /**
+     * {@inheritdoc}
+     * @param SecurityCondition $object
```
Comment (dantleech) [inline src/Sulu/Component/Security/Authorization/AccessControl/AccessControlVoter.php]: If `{@inheritDoc}` then no `@param`

File: src/Sulu/Component/Security/Authorization/AccessControl/AccessControlVoter.php [hunk 19]
```diff
@@ -0,0 +1,79 @@
...
+        if ($object->getObjectType() === null || $object->getObjectId() === null) {
+            return VoterInterface::ACCESS_ABSTAIN;
+        }
+
+        try {
+            $objectIdentity = new ObjectIdentity($object->getObjectId(), $object->getObjectType());
+            $this->aclProvider->findAcl($objectIdentity); // only called to check if acl exists
+
+            return parent::vote($token, $objectIdentity, $attributes);
+        } catch (AclNotFoundException $exc) {
```
Comment (dantleech) [inline src/Sulu/Component/Security/Authorization/AccessControl/AccessControlVoter.php]: `s/$exc/$e`

File: src/Sulu/Component/Security/Authorization/AccessControl/SymfonyAccessControlManager.php [hunk 20]
```diff
@@ -0,0 +1,98 @@
...
+
+use Sulu\Component\Security\Authentication\SecurityIdentityInterface;
+use Sulu\Component\Security\Authorization\MaskConverterInterface;
+use Symfony\Component\Security\Acl\Domain\ObjectIdentity;
+use Symfony\Component\Security\Acl\Domain\RoleSecurityIdentity;
+use Symfony\Component\Security\Acl\Exception\AclNotFoundException;
+use Symfony\Component\Security\Acl\Model\EntryInterface;
+use Symfony\Component\Security\Acl\Model\MutableAclProviderInterface;
+
+class SymfonyAccessControlManager implements AccessControlManagerInterface
```
Comment (dantleech) [inline src/Sulu/Component/Security/Authorization/AccessControl/SymfonyAccessControlManager.php]: Missing doc .. what is the reason for this class?

File: src/Sulu/Component/Security/Authorization/SecurityCondition.php [hunk 21]
```diff
@@ -0,0 +1,92 @@
...
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Security\Authorization;
+
+/**
+ * The class which describes the necessary permissions to access a certain element
+ * @package Sulu\Bundle\SecurityBundle\Security
```
Comment (dantleech) [inline src/Sulu/Component/Security/Authorization/SecurityCondition.php]: Remove package

File: src/Sulu/Component/Security/Authorization/SecurityCondition.php [hunk 22]
```diff
@@ -0,0 +1,92 @@
...
+ */
+class SecurityCondition
+{
+    /**
+     * The string representation of the security context
```
Comment (dantleech) [inline src/Sulu/Component/Security/Authorization/SecurityCondition.php]: These descriptions are duplicated three times in this class. I would only document the public API

File: src/Sulu/Component/Security/Authorization/SecurityCondition.php [hunk 23]
```diff
@@ -0,0 +1,92 @@
...
+     * Returns the id of the object
+     * @return mixed
+     */
+    public function getObjectId()
+    {
+        return $this->objectId;
+    }
+
+    /**
+     * Returns the locale in which the security
```
Comment (dantleech) [inline src/Sulu/Component/Security/Authorization/SecurityCondition.php]: `in which the security .. ?`

File: src/Sulu/Component/Security/Authorization/SecurityContextVoter.php [hunk 24]
```diff
@@ -0,0 +1,223 @@
...
+     */
+    public function supportsAttribute($attribute)
+    {
+        return in_array($attribute, array_keys($this->permissions));
+    }
+
+    /**
+     * Checks if the voter supports the given class.
+     *
+     * @param string $class A class name
```
Comment (dantleech) [inline src/Sulu/Component/Security/Authorization/SecurityContextVoter.php]: We should standardize on having an empty line or not having an empty line before `@params`. I would vote for an empty line as this is the defacto standard in Symfony

File: src/Sulu/Component/Security/Authorization/SecurityContextVoter.php [hunk 25]
```diff
@@ -0,0 +1,223 @@
...
+        }
+    }
+
+    /**
+     * Returns the vote for the given parameters.
+     *
+     * This method must return one of the following constants:
+     * ACCESS_GRANTED, ACCESS_DENIED, or ACCESS_ABSTAIN.
+     *
+     * @param TokenInterface $token      A TokenInterface instance
```
Comment (dantleech) [inline src/Sulu/Component/Security/Authorization/SecurityContextVoter.php]: Inconsistenty spacing. Use single space.

File: tests/Sulu/Component/Security/Authorization/AccessControl/SymfonyAccessControlManagerTest.php [hunk 26]
```diff
@@ -0,0 +1,188 @@
...
+    {
+        parent::setUp();
+
+        $this->aclProvider = $this->prophesize(MutableAclProviderInterface::class);
+
+        $this->maskConverter = $this->prophesize(MaskConverterInterface::class);
+
+        $this->securityIdentity = new RoleSecurityIdentity('SULU_ROLE_ADMINISTRATOR');
+
+        $this->acl = $this->prophesize(MutableAclInterface::class);
```
Comment (dantleech) [inline tests/Sulu/Component/Security/Authorization/AccessControl/SymfonyAccessControlManagerTest.php]: Remove empty lines

File: tests/Sulu/Component/Security/Authorization/SecurityContextVoterTest.php [hunk 27]
```diff
@@ -0,0 +1,270 @@
...
+class SecurityContextVoterTest extends ProphecyTestCase
+{
+    protected $permissions = array(
+        'view' => 64,
+        'add' => 32,
+        'edit' => 16,
+        'delete' => 8,
+        'archive' => 4,
+        'live' => 2,
+        'security' => 1
```
Comment (dantleech) [inline tests/Sulu/Component/Security/Authorization/SecurityContextVoterTest.php]: Not related, but making the values of the these configurable is dangerous. They should at least be validated somewhere.

**Comment by dantleech** [comment]
Generally good. You need to remove the `ProphecyTestCase` and replace them with `PHPUnit_Frameowkr_Testcase` for the tests to pass.

Would be good for @wachterjohannes to review also as my knowledge of security isn't so great.

File: src/Sulu/Bundle/SecurityBundle/Controller/PermissionController.php (hunk 4 above)
Comment (danrot) [inline src/Sulu/Bundle/SecurityBundle/Controller/PermissionController.php]: I think @chirimoya decided some time ago that it is `$exc` ;-)

File: src/Sulu/Bundle/AdminBundle/Navigation/ContentNavigationItem.php (hunk 5 above)
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Navigation/ContentNavigationItem.php]: Generally yes, but in this case I am injecting more options to be merged here... I don't think I can add additional options using the serializer, or am I wrong?

File: src/Sulu/Bundle/ContentBundle/Controller/NodeController.php (hunk 6 above)
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Controller/NodeController.php]: Hm, if I set the intellij formatter to the symfony2 conventions it does that for me... I think we should stick with it then, since most people might have configured it like that.

File: src/Sulu/Bundle/AdminBundle/Navigation/ContentNavigationItem.php (hunk 5 above)
Comment (dantleech) [inline src/Sulu/Bundle/AdminBundle/Navigation/ContentNavigationItem.php]: Not without some thought at least -- there is no obvious way that I am aware of.

File: src/Sulu/Bundle/ContentBundle/Controller/NodeController.php (hunk 7 above)
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Controller/NodeController.php]: Because we don't have an object at the time this is called. Or am I overlooking something?

File: src/Sulu/Bundle/SecurityBundle/Controller/PermissionController.php (hunk 9 above)
Comment (danrot) [inline src/Sulu/Bundle/SecurityBundle/Controller/PermissionController.php]: Just tested it, if I remove the try catch block I get a 500. Why should that be handled automatically?

File: src/Sulu/Bundle/SecurityBundle/Controller/PermissionController.php (hunk 10 above)
Comment (danrot) [inline src/Sulu/Bundle/SecurityBundle/Controller/PermissionController.php]: Same as above

File: src/Sulu/Bundle/SecurityBundle/Command/CreateRoleCommand.php (hunk 8 above)
Comment (danrot) [inline src/Sulu/Bundle/SecurityBundle/Command/CreateRoleCommand.php]: I'll set it to `127` for now. As long as there are only 7 permissions this should be enough, and it's not really part of this PR...

File: src/Sulu/Bundle/SecurityBundle/Entity/User.php (hunk 13 above)
Comment (danrot) [inline src/Sulu/Bundle/SecurityBundle/Entity/User.php]: Hm, I can't imagine of any... And apart from that the `getRoles` method is defined in Symfony, so no chance to put a parameter there.

File: src/Sulu/Bundle/ContentBundle/Controller/NodeController.php (hunk 7 above)
Comment (dantleech) [inline src/Sulu/Bundle/ContentBundle/Controller/NodeController.php]: I would intuitively expect `$hasPermission = $checker->checkObject($object)` type of API. Without the object maybe `$checker->checkClassId(Structure::class, $id)`.

I might then check the permission to access the object via. EventListeners at the storage layer.

File: src/Sulu/Bundle/SecurityBundle/Resources/public/js/components/permission-tab/components/form/main.js (hunk 15 above)
Comment (danrot) [inline src/Sulu/Bundle/SecurityBundle/Resources/public/js/components/permission-tab/components/form/main.js]: That means we have to pass all the available symfony routes to the client somehow... I am not sure how this can be solved easily...

File: src/Sulu/Component/Security/Authorization/AccessControl/AccessControlManagerInterface.php (hunk 17 above)
Comment (danrot) [inline src/Sulu/Component/Security/Authorization/AccessControl/AccessControlManagerInterface.php]: Yeah, it is, I'll remove it in the configuration now...

File: src/Sulu/Component/Security/Authorization/AccessControl/AccessControlVoter.php (hunk 19 above)
Comment (danrot) [inline src/Sulu/Component/Security/Authorization/AccessControl/AccessControlVoter.php]: Same as above

File: src/Sulu/Component/Security/Authorization/SecurityCondition.php (hunk 22 above)
Comment (danrot) [inline src/Sulu/Component/Security/Authorization/SecurityCondition.php]: I removed the description from the constructor, but I would leave them at the variable. If you use an IDE and jump to these variables you don't have to look up getter to see what's happening here.

File: src/Sulu/Component/Security/Authorization/SecurityContextVoter.php (hunk 24 above)
Comment (danrot) [inline src/Sulu/Component/Security/Authorization/SecurityContextVoter.php]: That's probably why it is different here... Gonna replace them with `{@inheritdoc}`, since they are not touched.

File: src/Sulu/Component/Security/Authorization/SecurityContextVoter.php (hunk 25 above)
Comment (danrot) [inline src/Sulu/Component/Security/Authorization/SecurityContextVoter.php]: Same as above, I'll use `{@inheritdoc}` here.

File: src/Sulu/Bundle/SecurityBundle/Resources/public/js/components/permission-tab/components/form/main.js (hunk 15 above)
Comment (dantleech) [inline src/Sulu/Bundle/SecurityBundle/Resources/public/js/components/permission-tab/components/form/main.js]: That should be the easy part .. we just need to generate all the admin routes and encode them in a JSON array. The only "issue" would be if we want to support parameter substitution, but that is also quite easily solvable.

Note there is also the https://github.com/FriendsOfSymfony/FOSJsRoutingBundle which I generally don't like, but it might be a good fit here.

File: src/Sulu/Bundle/ContentBundle/Controller/NodeController.php (hunk 7 above)
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Controller/NodeController.php]: By storage layer you mean doctrine? Then we couple to that, what I would not prefer... What if you don't use doctrine to store this data? That wouldn't work then anymore.

File: src/Sulu/Bundle/SecurityBundle/Resources/public/js/components/permission-tab/components/form/main.js (hunk 15 above)
Comment (danrot) [inline src/Sulu/Bundle/SecurityBundle/Resources/public/js/components/permission-tab/components/form/main.js]: Not sure if I should like that. We couple the javascript to the route names then, whereby the URL shouldn't change anyway... I guess we won't do that in this PR, so lets discuss this another time.

**Comment by danrot** [comment]
@dantleech Fixed all the stuff. @wachterjohannes is also not soo familiar with the symfony security stuff I think, but maybe he wants to have a look as well.

Did you also test this stuff on a real installation?

**Comment by dantleech** [comment]
As discussed this could also be implemented at the Storage layer using event subscribers, this has the added benefit that objects are secured globally and security is then not coupled to the controllers. We said that perhaps both systems could co-exist and that creation or migration to a storage layer solution should not cause a great BC break.

**Comment by dantleech** [comment]
Need to initialize the ACL in the build

**Comment by danrot** [comment]
No need to initialize the ACL in the build, since it is coupled to the configuration and then the tables are created by the `doctrine:schema:update` command.
